5.sc_Morphology_Logistic_Regression/0.split_data/scripts/0.split_data_binary_classification.py

```diff
@@ -0,0 +1,251 @@
+from sklearn.utils import parallel_backend, shuffle
+
+# In[2]:
+
+
+# Parameters
+cell_type = "SHSY5Y"
+aggregation = True
+nomic = True
+flag = True
+control = "DMSO_0.100_%_DMSO_0.025_%"
+treatment = "LPS_100.000_ug_per_ml_DMSO_0.025_%"
+
+
+# In[3]:
+
+
+MODEL_TYPE = "binary_classification"
+if flag == False:
+    # read in toml file and get parameters
+    toml_path = pathlib.Path("../1.train_models/single_class_config.toml")
+    with open(toml_path, "r") as f:
+        config = toml.load(f)
+    f.close()
+    control = config["logistic_regression_params"]["control"]
+    treatment = config["logistic_regression_params"]["treatments"]
+    aggregation = ast.literal_eval(config["logistic_regression_params"]["aggregation"])
+    nomic = ast.literal_eval(config["logistic_regression_params"]["nomic"])
+    cell_type = config["logistic_regression_params"]["cell_type"]
+    print(aggregation, nomic, cell_type)
+
+
+# In[4]:
+
+
+path = pathlib.Path(f"../../data/{cell_type}_preprocessed_sc_norm.parquet")
+
+data_df = pd.read_parquet(path)
+
+data_df.head(2)
+
+
+# In[5]:
+
+
+data_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"].unique()
+
+
+# In[6]:
+
+
+# set save path
+if aggregation == True:
+    if nomic == True:
+        save_path = pathlib.Path(
+            f"./indexes/{cell_type}/{MODEL_TYPE}/{control}_{treatment}"
+        )
+    elif nomic == False:
+        save_path = pathlib.Path(
+            f"./indexes/{cell_type}/{MODEL_TYPE}{control}_{treatment}"
+        )
+elif aggregation == False:
+    if nomic == True:
+        save_path = pathlib.Path(
+            f"./indexes/{cell_type}/{MODEL_TYPE}/{control}_{treatment}"
+        )
+    elif nomic == False:
+        save_path = pathlib.Path(
+            f"./indexes/{cell_type}/{MODEL_TYPE}/{control}_{treatment}"
+        )
+else:
+    print("Error")
+
+print(save_path)
+# create save path if it doesn't exist
+save_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[ ]:
+
+
+if nomic == True:
+    nomic_df_path = pathlib.Path(
+        f"../../2.Nomic_nELISA_Analysis/Data/clean/Plate2/nELISA_plate_430420_{cell_type}.csv"
+    )
+    df_nomic = pd.read_csv(nomic_df_path)
+    # drop columns that contain [pgML]
+    df_nomic = df_nomic.drop(
+        columns=[col for col in df_nomic.columns if "[pgML]" in col]
+    )
+    # drop first 25 columns
+    df_nomic = df_nomic.drop(columns=df_nomic.columns[3:25])
+    df_nomic = df_nomic.drop(columns=df_nomic.columns[0:2])
+else:
+    df_nomic = None
+
+
+# In[ ]:
+
+
+# subset each column that contains metadata
+metadata = data_df.filter(regex="Metadata")
+
+# get all columns that are not metadata except for metadata_Well
+data = data_df.drop(metadata.columns, axis=1)
+
+# get the metadata_Well column
+metadata_well = metadata["Metadata_Well"]
+
+data = pd.merge(data, metadata_well, left_index=True, right_index=True)
+
+
+# In[ ]:
+
+
+if (aggregation == True) and (nomic == True):
```

Comment:
```suggestion
if not aggregation and nomic:
```

Reply:
addressed and changed thank you for correcting!